src/__main__.py

```diff
@@ -55,6 +62,40 @@ def quit_server():
     server.stop()
     return 'good-bye'
 
+def allowed_file(filename):
+    return '.' in filename and \
+           filename.rsplit('.', 1)[1].lower() in ALLOWED_EXTENSIONS
+
+@app.route('/', methods=['GET', 'POST'])
+def upload_file():
+    if request.method == 'POST':
+        # check if the post request has the file part
+        if 'file' not in request.files:
+            flash('No file part')
+            return redirect(request.url)
+        file = request.files['file']
+        # if user does not select file, browser also
+        # submit an empty part without filename
+        if file.filename == '':
+            flash('No selected file')
+            return redirect(request.url)
+        if file and allowed_file(file.filename):
+            filename = secure_filename(file.filename)
+            file.save(os.path.join(app.config['UPLOAD_FOLDER'], filename))
+            return render_template('index.html')
+    return '''
+    <!doctype html>
+    <title>Upload new File</title>
+    <h1>Upload new File</h1>
+    <form method=post enctype=multipart/form-data>
+      <input type=file name=file>
+      <input type=submit value=Upload>
+    </form>'''
+
+@app.route('/uploads/<filename>')
+def uploaded_file(filename):
```

Comment:
This should probably be rerouted to `/downloads/<filename>`, since the web client is set up from the POV of the mobile user who will be _downloading_ these files from the server.

Reply:
forgot to remove this, will test it again to make sure i dont need this part and then will remove it